File: 1. Multiples of 3 or 5/solution.go
package main

import "fmt"

func sumMultiples(limit int) int {
	sumDivisibleBy := func(divisor, limit int) int {
		n := (limit - 1) / divisor
		return divisor * (n * (n + 1)) / 2
	}

	sum := sumDivisibleBy(3, limit) + sumDivisibleBy(5, limit) - sumDivisibleBy(15, limit)
	return sum
}

func main() {
	limit := 1000
	result := sumMultiples(limit)
	fmt.Println(result)
}

File: 1. Multiples of 3 or 5/solution.ts
function sumMultiples(limit: number): number {
    const sumDivisibleBy = (divisor: number) => {
        const n = Math.floor((limit - 1) / divisor);
        return divisor * (n * (n + 1)) / 2;
    };

    const sum = sumDivisibleBy(3) + sumDivisibleBy(5) - sumDivisibleBy(15);
    return sum;
}

const limit = 1000;
const result = sumMultiples(limit);
console.log(result);